File: testprojects/master_python_mainone/pipelines/PYTHON_DEP_MANAGEMENT_ALL/code/job/graph/FlattenSchema_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.libs import typed_lit
from job.config.ConfigStore import *
from job.udfs.UDFs import *

def FlattenSchema_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn("c_array-- boolean ", explode_outer("c_array-- boolean "))\
        .withColumn("c_array-string  _ string", explode_outer("c_array-string  _ string"))\
        .withColumn("c_array-int  _ int", explode_outer("c_array-int  _ int"))\
        .withColumn("c_array--long", explode_outer("c_array--long"))\
        .select(col("c_array-int  _ int"), col("c_array-string  _ string"), col("c_array--long"), col("c_array-- boolean "), col("p_short"), col("p_int"), col("c_decimal  -  "))

File: testprojects/scala_main/pipelines/DONOT_OPEN_SCALA/code/src/main/scala/org/main/scla_dep_mgmt/graph/FlattenSchema_1.scala
package org.main.scla_dep_mgmt.graph

import io.prophecy.libs._
import org.main.scla_dep_mgmt.udfs.UDFs._
import org.main.scla_dep_mgmt.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object FlattenSchema_1 {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.withColumn("c_array_int",       explode_outer(col("c_array_int")))
      .withColumn("c_array_string",    explode_outer(col("c_array_string")))
      .withColumn("c_array_long",      explode_outer(col("c_array_long")))
      .withColumn("c_array_boolean",   explode_outer(col("c_array_boolean")))
      .withColumn("c_array_date",      explode_outer(col("c_array_date")))
      .withColumn("c_array_timestamp", explode_outer(col("c_array_timestamp")))
      .withColumn("c_array_float",     explode_outer(col("c_array_float")))
      .withColumn("c_array_decimal",   explode_outer(col("c_array_decimal")))
      .withColumn("c_struct-c_array_int",
                  explode_outer(col("c_struct.c_array_int"))
      )
      .select(
        col("c_array_int"),
        col("c_array_string"),
        col("c_array_long"),
        col("c_array_boolean"),
        col("c_array_date"),
        col("c_array_timestamp"),
        col("c_array_float"),
        col("c_array_decimal"),
        col("c_struct.c_array_int").as("c_struct-c_array_int"),
        col("c_struct.c_timestamp").as("c_struct-c_timestamp"),
        col("c_struct.c_short").as("c_struct-c_short"),
        col("c_struct.c_decimal").as("c_struct-c_decimal")
      )

}
